middleware/apimiddleware.go

```diff
@@ -177,6 +178,33 @@ func GetRawHexTx(r *http.Request) (string, error) {
 	return rawHexTx, nil
 }
 
+// OriginalRequestURI checks the X-Original-Request-URI HTTP request header for
+// a valid URI, and patches the request URL's Path and RawPath. This may be
+// useful in the event that a reverse proxy maps requests on path A to path B,
+// but the request handler requires the original path A (e.g. generating links
+// relative to path A).
+func OriginalRequestURI(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origRequestURI := r.Header.Get("X-Original-Request-URI")
```

Comment:
Note that without `X-Original-Request-URI`, the page is still served, but if the downstream handlers use `r.URL.Path` (or `r.URL.RequestURI()`), the path may be different from the client's original request URI if a proxy is changing the request headers on the way to the dcrdata backend.